apps/mobile/src/components/welcome-screen/index.tsx

```diff
@@ -146,6 +147,16 @@ export function WelcomeScreen() {
     };
   }, []);
 
+  // Show checkmark for 800ms before changing to the next screen state.
+  useEffect(() => {
+    if (isSubmitted) {
+      setShowSuccess(true);
+      setTimeout(() => {
+        setShowSuccess(false);
+      }, 800);
+    }
+  }, [isSubmitted]);
```

Comment:
```suggestion
  useEffect(() => {
    if (isSubmitted) {
      setShowSuccess(true);
      const timeout = setTimeout(() => {
        setShowSuccess(false);
      }, 800);
      return () => clearTimeout(timeout);
    }
  }, [isSubmitted]);
```

Reply:
probs not necessary as it's a single page, but good practice to avoid the react runtime error